src/OrbitGl/ThreadTrack.cpp

```diff
@@ -446,11 +446,25 @@ void ThreadTrack::OnCaptureComplete() {
 }
 
 [[nodiscard]] static inline uint64_t GetNextPixelBoundaryTimeNs(
-    float world_x, const internal::DrawData& draw_data) {
-  float normalized_x = (world_x - draw_data.track_start_x) / draw_data.track_width;
-  int pixel_x = static_cast<int>(
-      ceil(normalized_x * draw_data.viewport->WorldToScreenWidth(draw_data.track_width)));
-  return draw_data.min_tick + pixel_x * draw_data.ns_per_pixel;
+    uint64_t current_end_time, const internal::DrawData& draw_data) {
+  uint64_t ns_from_min = current_end_time - draw_data.min_tick;
+  uint64_t total_ns = draw_data.max_tick - draw_data.min_tick;
+  uint64_t num_pixels_on_track = draw_data.viewport->WorldToScreenWidth(draw_data.track_width);
+
+  // If max_tick and min_tick are the same, there is no way to calculate the next pixel, as every
+  // pixel has the same timestamp.
+  if (total_ns == 0) {
+    return current_end_time + 1;
+  }
```

Comment:
Shouldn't this return draw_data.max_tick? If all pixels have the same timestamp, any timer for that timestamp will be drawn on the first iteration, we shouldn't draw anything else, no?

Reply:
Exactly. Currently the UI is completely broken (for example see http://b/168283058) but it doesn't crash at all. I'm not sure if we should fix or limit the zooming-level only to not have to address this border cases.

In this case, we probably will be inside the only timer that we should draw:
current_start_time <= min_tick = max_tick <= current_end_time, and we should draw only that timer.

The problem of returning max_tick is that we are returning a pixel to draw inside the drawable timer, so basically we are telling that we can draw this timer again. With my optimization this will result in an infinite loop, because I have a implicit invariant that this function should return a time greater than the current_end_time provided.